### PR TITLE
Update `execa` to `6.0.0`

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

It's a pure ESM package.

ref: https://github.com/sindresorhus/execa/releases/tag/v6.0.0

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/config/install-prettier.js [hunk 1]
```diff
@@ -14,15 +14,15 @@ if (!allowedClients.has(client)) {
 
 module.exports = (packageDir) => {
   const tmpDir = tempy.directory();
-  const fileName = execa
-    .sync("npm", ["pack"], { cwd: packageDir })
-    .stdout.trim();
+  const fileName = execaSync("npm", ["pack"], {
```
Comment (fisker) [inline tests/config/install-prettier.js:17]: Did you try use it directly? This file should be able to use ESM.

Comment (fisker) [reply tests/config/install-prettier.js:17]: Never mind, I forgot we need run tests on Node.js 10

**Review by fisker — APPROVED** [review]

File: tests/config/install-prettier.js (hunk 1 above)
Comment (sosukesuzuki) [reply tests/config/install-prettier.js:17]: I've noticed Node.js 10 doesn't support `"imports"` in `package.json`. I'll consider the other way.

**Comment by fisker** [comment]
Use `await import(vendor)` to find specifiers?

**Pull request merged**